knotx-core/knotx-service-adapter-http/src/test/java/com/cognifide/knotx/core/serviceadapter/http/HttpServiceAdapterTest.java

```diff
@@ -0,0 +1,121 @@
+
+  //Test Runner Rule of Verts
+  private RunTestOnContext vertx = new RunTestOnContext();
+
+  //Test Runner Rule of Knotx
+  private TestVertxDeployer knotx = new TestVertxDeployer(vertx);
+
+  //Junit Rule, sets up logger, prepares verts, starts verticles according to the config (supplied in annotation of test method)
+  @Rule
+  public RuleChain chain = RuleChain.outerRule(new Logback()).around(vertx).around(knotx);
+
+  @Test
+  @KnotxConfiguration("knotx-service-adapter-http-test.json")
+  public void callNonExistingService_expectBadRequestResponse(TestContext context) {
+    callAdapterServiceWithAssertions(context, "not/existing/service/address", HttpMethod.GET, httpResponseWrapper ->
+        context.assertTrue(httpResponseWrapper.statusCode().equals(HttpResponseStatus.BAD_REQUEST)));
+  }
+
+
+  @Test
+  @KnotxConfiguration("knotx-service-adapter-http-test.json")
+  public void callExistingServiceUsingGETMethod_expectOKResponseWithServiceDataProvidedByService1(TestContext context) {
+    callAdapterServiceWithAssertions(context, "/service/mock/first.json", HttpMethod.GET, httpResponseWrapper -> {
+      context.assertTrue(httpResponseWrapper.statusCode().equals(HttpResponseStatus.OK));
+      context.assertTrue(httpResponseWrapper.headers().isEmpty());
+
+      try {
+        JsonObject serviceResponse = new JsonObject(httpResponseWrapper.body().toString());
+        JsonObject expectedResponse = new JsonObject(FileReader.readText("first-response.json"));
+        context.assertEquals(serviceResponse, expectedResponse);
+      } catch (Exception e) {
+        context.fail(e);
+      }
+    });
+  }
+
+  @Test
+  @KnotxConfiguration("knotx-service-adapter-http-test.json")
+  public void callExistingServiceUsingPOSTMethod_expectOKResponseWithServiceDataProvidedByService1(TestContext context) {
+    callAdapterServiceWithAssertions(context, "/service/mock/first.json", HttpMethod.POST, httpResponseWrapper -> {
+      context.assertTrue(httpResponseWrapper.statusCode().equals(HttpResponseStatus.OK));
+      context.assertTrue(httpResponseWrapper.headers().isEmpty());
+
+      try {
+        JsonObject serviceResponse = new JsonObject(httpResponseWrapper.body().toString());
+        JsonObject expectedResponse = new JsonObject(FileReader.readText("first-response.json"));
+        context.assertEquals(serviceResponse, expectedResponse);
+      } catch (Exception e) {
+        context.fail(e);
+      }
+    });
+  }
+
+  private void callAdapterServiceWithAssertions(TestContext context, String servicePath, HttpMethod method, Action1<HttpResponseWrapper> testFunction) {
+    JsonObject message = getPayloadMessage(servicePath, method);
+
+    Async async = context.async();
+
+    vertx.vertx().eventBus().<JsonObject>send(ADAPTER_ADDRESS, message, ar -> {
+      if (ar.succeeded()) {
+        Observable
+            .just(new HttpResponseWrapper(ar.result().body()))
+            .doOnNext(testFunction);
```

Comment:
I think you're missing subscribe method otherwise  observable will not generate anything. Can be simply subscribe() - were not interested in valuesame.